test/e2e/clusterclass_changes.go

```diff
@@ -738,13 +819,47 @@ func rebaseClusterClassAndWait(ctx context.Context, input rebaseClusterClassAndW
 		}, input.WaitForMachineDeployments...).Should(BeNil())
 	}
 
-	// Verify that the ControlPlane has not been changed.
-	// NOTE: MachineDeployments are rolled out before the ControlPlane. Thus, we know that the
-	// ControlPlane would have been updated by now, if there have been any changes.
-	afterControlPlane, err := external.Get(ctx, mgmtClient, controlPlaneRef)
+	if input.ControlPlaneChanged {
+		Eventually(func(g Gomega) {
+			afterControlPlane, err := external.Get(ctx, mgmtClient, controlPlaneRef)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(afterControlPlane.GetGeneration()).ToNot(Equal(beforeControlPlane.GetGeneration()),
+				"ControlPlane generation should be incremented during the rebase because ControlPlane expected to be changed.")
+		}, input.WaitForMachineDeployments...).Should(BeNil())
```

Comment:
```suggestion
		}, input.WaitForMachineDeployments...).Should(Succeed())
```
Same in l.838

BeNil sounds strange if we don't have a return parameter

Probably behaves the same way as BeNil here, but not sure (the implementation of these matchers is slightly different)